### ISSUE TITLE
Plugin becomes unresponsive and locks up Homebridge

### ISSUE THREAD

**Issue by AlexGustafsson** [plasticrake/homebridge-tplink-smarthome#14]
I've dealt with some issues where one of my plugs would turn of and become unresponsive. When this happens, naturally the plugin can't reach the plug. Due to this it throws a TCP error (well the dependency hs100-api throws an error). The socket/TCPerror, `EHOSTUNREACH` seems to clog up the systems. Whenever the error is thrown, the plugin as well as all other plugins become unresponsive. It seems to hang. Whenever the connection is reestablished, it seems to connect to the plug again, logs that the plug is online and then lets other plugins I have do their thing.

So, TLDR;
Whenever the TCP error `EHOSTUNREACH`is thrown, the plugin freezes homebridge until it can connect to the plug again.

**Comment by AlexGustafsson** [comment]
Also related: #12

**Comment by plasticrake** [comment]
Mostly fixed in 2.0.1 just released. Should now be properly updating to the new IP address when reading from the cache which should minimize this issue. I haven't fully tested other connection issues or how to recover, so I'll leave this issue open for now.

**Comment by plasticrake** [comment]
Some more updates in 2.2.0 to properly handle IP address changes

**Comment by thusithapk** [comment]
Issue is still happening and causing Homebridge to stop working properly.
Reproduce:
1) Setup Homebridge with 2 x HS100 plugs
2) Test everything and confirm working as expected
3) Remove one HS100 plug from the wall socket
4) Try Home app and see

If you can fix this would be really good otherwise whole Homebridge will stop working.

**Comment by AlexGustafsson** [comment]
I'll try to do some "field testing" during the next few days to see if the 3.1.0 has brought greater reliability. So far it seems to have solved the issues.

**Comment by plasticrake** [comment]
@AlexGustafsson 3.1.0 is now properly calling the homebridge callback when there is an issue contacting a TPLink device so it shouldn't make everything fail. I did some testing and before it would break everything, and now it seems to recover nicely.

**Comment by thusithapk** [comment]
Works well.
Only thing is that when a plug recovers (online again) all devices go "No Response" for few seconds before all comes back online.
Would you be able to make the recovery quicker?
